lib/minitest/stub_on_roids.rb

```diff
@@ -22,11 +22,19 @@ def stub_with_args(name, val_or_callable, expected_args, &block)
       end
     end
 
-    def stub_and_expect(name, val_or_callable = nil, expected_args = [], times: 1)
+    def stub_and_expect(name, val_or_callable = nil, expected_args = [], times: 1, expectations: [])
       raise MethodAlreadyStubbedError, "Method :#{name} already stubbed" if self.respond_to? "__minitest_stub__#{name}"
       
       mock = Minitest::Mock.new
-      times.times { mock.expect :call, val_or_callable, expected_args }
+
+      if expectations.size == 0
+        times.times { mock.expect :call, val_or_callable, expected_args }
+      else
+        expectations.each do |expectation|
+          [:expected_args, :returned_value].each { |k| raise ArgumentError, "#{k} not found in times_expectations param" if expectation[k] == nil }
```

Comment:
```suggestion
          [:expected_args, :returned_value].each { |k| raise ArgumentError, "#{k} not found in times_expectations param" if expectation[k].nil? }
```

Reply:
1. Let's use `return_value` instead of `returned_value`.
2. "times_expectations param"?